tests/test_skills/test_abstract_round_abci/test_behaviours_utils.py

```diff
@@ -580,25 +610,52 @@ def test_async_act_wrapper_communication(
             else:
                 _log_end_mock.assert_not_called()
 
+    @pytest.mark.parametrize(
+        "expected_round_count, expected_reset_index",
+        ((None, None), (0, 0), (123, 4), (235235, 754)),
+    )
     @mock.patch.object(
         BehaviourATest,
         "_BaseBehaviour__check_tm_communication",
         side_effect=yield_and_return_bool_wrapper(True),
     )
-    @mock.patch.object(BaseBehaviour, "_get_status", _get_status_patch)
-    def test_async_act_wrapper_agent_sync_mode(self, _: mock._patch) -> None:
+    def test_async_act_wrapper_agent_sync_mode(
+        self,
+        _: mock._patch,
+        expected_round_count: Optional[int],
+        expected_reset_index: Optional[int],
+    ) -> None:
         """Test 'async_act_wrapper' in sync mode."""
         self.behaviour.context.state.round_sequence.syncing_up = True
         self.behaviour.context.state.round_sequence.height = 0
         self.behaviour.matching_round = MagicMock()
         self.behaviour.context.logger.info = lambda msg: logging.info(msg)  # type: ignore
 
-        with mock.patch.object(logging, "info") as log_mock:
+        with mock.patch.object(logging, "info") as log_mock, mock.patch.object(
+            BaseBehaviour,
+            "_get_status",
+            _get_status_patch_wrapper(0, expected_round_count, expected_reset_index),
+        ):
             gen = self.behaviour.async_act_wrapper()
-            try_send(gen)
-            try_send(gen)
+            for __ in range(3):
+                try_send(gen)
```

Comment:
why 3?

Reply:
Notice how the `yield` was incorrectly placed [here](https://github.com/valory-xyz/open-autonomy/pull/1049/files?file-filters%5B%5D=.py&show-viewed-files=true#diff-63b131ca6614d6666a3eeb925cb9b9e71b63f27a68ea748d9db7f8bae21f0bc5L375-L378). Now it is placed before the `return`, is reachable, and another `try_send` is required.